### PR TITLE
255 save load button

### PR DESCRIPTION
**Pull request by Untrustedlife**
It is done.

### CODE REVIEW

File: scripts/main_menu/main_menu_hud.lua [hunk 1]
```diff
@@ -7,17 +7,32 @@ end
...
 
 function MainMenuHudSystem:update(renderTime, logicTime)
+    if Engine:fileExists("quick.sav") then
+	    --print("The file exists");
+        root:getChild("Background"):getChild("MainMenuInteractive"):getChild("LoadGameButton"):enable();
+    else
+		--print("the file doesnt exist");
+        root:getChild("Background"):getChild("MainMenuInteractive"):getChild("LoadGameButton"):disable();
+    end
+end
```
Comment (jjonj) [inline scripts/main_menu/main_menu_hud.lua]: Seems a bit heavy to do IO every frame, could we do this in a smarter way?

Comment (Untrustedlife) [inline scripts/main_menu/main_menu_hud.lua]: We could make it only check when the user opens up the menu

File: scripts/microbe_stage/microbe_stage_hud.lua [hunk 2]
```diff
@@ -101,6 +105,13 @@ function HudSystem:update(renderTime)
         newZVal = 80
     end
     offset.z = newZVal --]]
+
+    --make the load button activate, if and only if there actually  is a save file to load
+    if Engine:fileExists("quick.sav") then
+        self.rootGUIWindow:getChild("LoadGameButton"):enable();
+    else
+        self.rootGUIWindow:getChild("LoadGameButton"):disable();
+    end
```
Comment (jjonj) [inline scripts/microbe_stage/microbe_stage_hud.lua]: Same comment as before, we would essentially be asking the operating system 60 times a second whether this file exist.

Comment (Untrustedlife) [inline scripts/microbe_stage/microbe_stage_hud.lua]: Like I said, we could make it only check when the user opens up the menu.

File: scripts/main_menu/main_menu_hud.lua (hunk 1 above)
Comment (jjonj) [inline scripts/main_menu/main_menu_hud.lua]: and also when a file is saved we would want to update availability but yeah something like that sounds good

**Comment by jjonj** [comment]
Hrmm I just tested the branch and the load button in the main menu worked great, but the load/save buttons in the microbe stage stayed grey regardless of their being a save file or not. The load button did however work despite being grey.
If you need help fixing that lemme know.

**Comment by Untrustedlife** [comment]
Your layout file isn't updated. Works fine on my end (the save + load buttons in yours font have the correct image set.

**Comment by Untrustedlife** [comment]
This is the reason I wanted to commit my layout file.

File: scripts/quick_save.lua [hunk 3]
```diff
@@ -15,6 +15,7 @@ function QuickSaveSystem:update(renderTime, logicTime)
         Engine:save("quick.sav")
     end
     if loadDown and not self.loadDown then
+        print("Loading")
```
Comment (Moopli) [inline scripts/quick_save.lua]: debug message

File: scripts/main_menu/main_menu_hud.lua [hunk 4]
```diff
@@ -7,17 +7,38 @@ end
...
     local microbeButton = root:getChild("Background"):getChild("MainMenuInteractive"):getChild("NewGameButton")
     local microbeEditorButton = root:getChild("Background"):getChild("MainMenuInteractive"):getChild("EditorMenuButton")
     local quitButton = root:getChild("Background"):getChild("MainMenuInteractive"):getChild("ExitGameButton")
-    
+    local loadButton = root:getChild("Background"):getChild("MainMenuInteractive"):getChild("LoadGameButton")   
     microbeButton:registerEventHandler("Clicked", mainMenuMicrobeStageButtonClicked)
     microbeEditorButton:registerEventHandler("Clicked", mainMenuMicrobeEditorButtonClicked)
+    loadButton:registerEventHandler("Clicked", mainMenuLoadButtonClicked)
     quitButton:registerEventHandler("Clicked", quitButtonClicked)
+	updateLoadButton();
```
Comment (Moopli) [inline scripts/main_menu/main_menu_hud.lua]: ze tabs

**Pull request closed without merging**